pandas/core/window/ewm.py

```diff
@@ -134,7 +134,8 @@ def _calculate_deltas(
     _times = np.asarray(
         times.view(np.int64), dtype=np.float64  # type: ignore[union-attr]
     )
-    _halflife = float(Timedelta(halflife).value)
+    # TODO: generalize to non-nano?
```

Comment:
Yeah this currently assumes `np.diff(_times)` and `_halflife` are both `ns`, so I think as long as they are both brought to the higher of the resolution it should be fine